lib/datadog/appsec/waf/context.rb

```diff
@@ -64,49 +66,33 @@ def run(persistent_data, ephemeral_data, timeout = LibDDWAF::DDWAF_RUN_TIMEOUT)
             coerce: false
           )
           if ephemeral_data_obj.null?
-            raise LibDDWAF::Error, "Could not convert ephemeral data: #{ephemeral_data.inspect}"
+            raise ConversionError, "Could not convert ephemeral data: #{ephemeral_data.inspect}"
           end
 
           result_obj = LibDDWAF::Result.new
-          raise LibDDWAF::Error, 'Could not create result object' if result_obj.null?
+          raise LibDDWAFError, "Could not create result object" if result_obj.null?
 
-          code = LibDDWAF.ddwaf_run(@context_obj, persistent_data_obj, ephemeral_data_obj, result_obj, timeout)
+          code = LibDDWAF.ddwaf_run(@context_ptr, persistent_data_obj, ephemeral_data_obj, result_obj, timeout)
 
-          result = Result.new(
+          Result.new(
             RESULT_CODE[code],
             Converter.object_to_ruby(result_obj[:events]),
             result_obj[:total_runtime],
             result_obj[:timeout],
             Converter.object_to_ruby(result_obj[:actions]),
             Converter.object_to_ruby(result_obj[:derivatives])
           )
-
-          [RESULT_CODE[code], result]
         ensure
           LibDDWAF.ddwaf_result_free(result_obj) if result_obj
           LibDDWAF.ddwaf_object_free(ephemeral_data_obj) if ephemeral_data_obj
         end
 
         private
 
-        # FIXME: Rename into something which reflect that it's impossible to run
-        #        libddwaf on finalized context (closed)
-        def validate!
-          @valid = true
-        end
-
-        def invalidate!
-          @valid = false
-        end
-
-        def valid?
-          @valid
-        end
-
-        def valid!
-          return if valid?
+        def ensure_pointer_presence!
+          return unless @context_ptr.nil?
```

Comment:
I think negation kills the purpose of the explicit `nil?` check

```suggestion
          return if @context_ptr
```